### ISSUE TITLE
Data Misinterpreted as Instructions

### ISSUE THREAD

**Issue by AlexanderAdelAU** [endlos99/xdt99#19]
I know this is probably a tiresome request, but I am wondering if a trace element could be added to the code (xdt99)  so that unreferenced location (where data and code are combined and there is no execution reference, i.e.  branch, jump etc) could be inferred as data.  At the moment much of the Data segments in the code are misinterpreted as instructions.  
This is especially important when you are trying to debug linked code.

Anyway, just a thought.

**Comment by endlos99** [comment]
You probably talk about `xda99`  and `xdg99`.

I fully agree with you, that separating data from code would be extremely helpful.  I didn't fully understand, however, what you are suggesting.

What would be an "unreferenced location" or "no execution reference"?  You mean words which are not jumps or branches?  Or words which are not jumped/branched to?

And what do you mean by "trace element"?  Just a marker to tell if some specific word is data or code?

**Comment by AlexanderAdelAU** [comment]
Firstly I apologise for the hurried text above.   But yes, an unreferenced location would be one where there was no execution reference which could be generalised as a location that was not accessible as a result of a JMP, Branch instruction and would probably include XOP.  This could be unequivocally defined as being data under normal programming practices.   So as you said 

> "Or words which are not jumped/branched to?"

 So, what I meant by the the `trace element`  was a 2nd pass that traced the disassembled programme to fix the data and code segments.  If it was not in Python, I would probably be able to do it for you :-)

**Comment by endlos99** [comment]
In a sense, `xda99` is already doing what you suggest if you use the run option `-r` instead of the from option `-f`.

But still, there are problems.  The most critical problem is that the disassembler can only follow direct branch instruction, but not indirect branches like `B *R1`.  Thus, it is not possible to identify _all_ code segments.

Another problem for `xdg99` is that the conditional `BR` is often used as a shorter unconditional branch `B`, which means that `xdg99` will disassemble two code branches, when the programmer only intended one branch.

In summary, disassembly cannot be accurate.

Usually, I use either disassembler with `-r`, and then analyze the result further to find or guess additional code segments, and then rerun the disassembly, and so on.

**Comment by AlexanderAdelAU** [comment]
Agree - 

> In summary, disassembly cannot be accurate.

Maybe we have to be thankful for what we have !
